pkg/operator/options/options_validation.go

```diff
@@ -31,11 +32,20 @@ func (o Options) Validate() error {
 		o.validateRequiredFields(),
 		o.validateEndpoint(),
 		o.validateVMMemoryOverheadPercent(),
+		o.validateNetworkPluginMode(),
 		o.validateVnetSubnetID(),
 		validate.Struct(o),
 	)
 }
 
+func (o Options) validateNetworkPluginMode() error {
+	// NOTE: Network Plugin Mode should be normalized on the AKS API Level, so if we are passing values in from the ManagedCluster, NetworkPluginMode will already be normalized.
+	if o.NetworkPluginMode != consts.PodNetworkTypeOverlay && o.NetworkPluginMode != "" && o.NetworkPluginMode != "none" {
```

Comment:
Yeah we can do that. I hesitated at first to do apimachinery.sets here because it's a bit much for 3 values, but there is virtually no cost to defining that grouping so let's do it.

Reply:
good point